modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/hooks/useEventListener.es.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {useEffect} from 'react';
+
+export default (target, type, listener, ...options) => {
+	useEffect(() => {
+		target.addEventListener(type, listener, ...options);
+
+		return () => {
+			target.removeEventListener(type, listener, ...options);
+		};
+	}, [target, type, listener, options]);
+};
```

Comment:
@alinedoleron, ideally, this code should be committed inside [this commit](https://github.com/natocesarrego/liferay-portal/pull/628/commits/b2d604f036658f3ef466b85bd0fa9c67a89a45b0) or be added before this mentioned commit, because the `useEventListener` is being invoked before its declaration and it can affect the reviewer's understanding. 😉 This is not a big deal, just keep it in mind for next pulls, ok?

Reply:
That's true. Sorry!

Reply:
No problem. 🙂